pandas/core/tools/datetimes.py

```diff
@@ -59,7 +60,7 @@
 # ---------------------------------------------------------------------
 # types used in annotations
 
-ArrayConvertible = Union[List, Tuple, ArrayLike, "Series"]
+ArrayConvertible = Union[List, Tuple, ArrayLike, "Series", MutableMapping]
```

Comment:
hmm, i would rather add this specifically where its needed as we often times don't actually accept a MutableMapping (yes we *could* but we have specific checks and this doesn't pass)

Reply:
Reverted back to original impl, so MutableMapping was no longer necessary